osu.Game.Modes.Taiko/Objects/Drawable/Pieces/CirclePiece.cs

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics.Backgrounds;
+using OpenTK.Graphics;
+using OpenTK;
+
+namespace osu.Game.Modes.Taiko.Objects.Drawable.Pieces
+{
+    /// <summary>
+    /// A circle piece which is used uniformly through osu!taiko to visualise hitobjects.
+    /// <para>
+    /// The body of this piece will overshoot it by Height/2 on both sides of its length, such that
+    /// a regular "circle" the result of setting Width to 0.
+    /// </para>
+    /// <para>
+    /// Hitobjects that have a length need only to set Width and the extra corner radius will be added internally.
+    /// </para>
+    /// </summary>
+    public abstract class CirclePiece : ScrollingCirclePiece

```

Comment:
this feels very very weird. `ScrollingCirclePiece` should probably just be `ScrollingPiece`?